alicloud/resource_alicloud_polardb_cluster.go

```diff
@@ -0,0 +1,426 @@
+		d.SetPartial("cluster_name")
+	}
+
+	if d.HasChange("security_ips") {
+		ipList := expandStringList(d.Get("security_ips").(*schema.Set).List())
+
+		ipstr := strings.Join(ipList[:], COMMA_SEPARATED)
+		// default disable connect from outside
+		if ipstr == "" {
+			ipstr = LOCAL_HOST_IP
+		}
+
+		if err := polarDBService.ModifyDBSecurityIps(d.Id(), ipstr); err != nil {
+			return WrapError(err)
+		}
+		d.SetPartial("security_ips")
+	}
+
+	d.Partial(false)
+	return resourceAlicloudPolarDBClusterRead(d, meta)
+}
+
+func resourceAlicloudPolarDBClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+
+	cluster, err := polarDBService.DescribePolarDBClusterAttribute(d.Id())
+	if err != nil {
+		if polarDBService.NotFoundCluster(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+
+	ips, err := polarDBService.GetSecurityIps(d.Id())
+	if err != nil {
+		return WrapError(err)
+	}
+
+	d.Set("security_ips", ips)
+
+	d.Set("cluster_network_type", cluster.DBClusterNetworkType)
+	d.Set("vswitch_id", cluster.VSwitchId)
+	d.Set("pay_type", cluster.PayType)
+	d.Set("id", cluster.DBClusterId)
+	d.Set("cluster_name", cluster.DBClusterDescription)
+	d.Set("db_type", cluster.DBType)
+	d.Set("db_version", cluster.DBVersion)
+	d.Set("maintain_time", cluster.MaintainTime)
+	d.Set("zone_ids", cluster.ZoneIds)
+	d.Set("cluster_charge_type", cluster.PayType)
+
+	if err = polarDBService.RefreshParameters(d, "parameters"); err != nil {
+		return WrapError(err)
+	}
+
+	if cluster.PayType == string(Prepaid) {
+		request := polardb.CreateDescribeAutoRenewAttributeRequest()
+		request.RegionId = client.RegionId
+		request.DBClusterIds = d.Id()
+
+		raw, err := client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.DescribeAutoRenewAttribute(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		response, _ := raw.(*polardb.DescribeAutoRenewAttributeResponse)
+		if response != nil && len(response.Items.AutoRenewAttribute) > 0 {
+			renew := response.Items.AutoRenewAttribute[0]
+			d.Set("auto_renew", renew.AutoRenewEnabled)
+			d.Set("auto_renew_period", renew.Duration)
+		}
+	}
+
+	return nil
+}
+
+func resourceAlicloudPolarDBClusterDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+
+	cluster, err := polarDBService.DescribePolarDBClusterAttribute(d.Id())
+	if err != nil {
+		if polarDBService.NotFoundCluster(err) {
+			return nil
+		}
+		return WrapError(err)
+	}
+	if PayType(cluster.PayType) == Prepaid {
+		return WrapError(Error("At present, 'Prepaid' cluster cannot be deleted and must wait it to be expired and release it automatically."))
```

Comment:
If the prepaid instance does not support deleted, please remove it from state, like https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_alikafka_instance.go#L419. Also,  you need to point it in the docs, like https://www.terraform.io/docs/providers/alicloud/r/alikafka_instance.html#removing-alicloud_alikafka_instance-from-your-configuration